File: py-combate-mortal/engine/unidad.py
class Unidad:
    def __init__(self, vida):
        self.vida = vida
        self.escudo = None

    def disparar(self, objetivo):
        objetivo.recibir_disparo(1)

    def recibir_disparo(self, danio):
        if self.escudo:
            danio = self.escudo.modificar_danio(danio)
        self.vida -= danio

    def esta_vivo(self):
        return self.vida > 0

    def agregar_escudo(self, escudo):
        self.escudo = escudo

File: combate-mortal/src/main/java/com/combatem/Unidad.java
package com.combatem;

public abstract class Unidad {
    protected double vida;
    protected Escudo escudo;

    public Unidad(double vida) {
        this.vida = vida;
        this.escudo = null;
    }

    public void disparar(Unidad objetivo) {
        objetivo.recibirDisparo(1);
    }

    public void recibirDisparo(double danio) {
        if (escudo != null) {
            danio = escudo.modificarDanio(danio);
        }
        vida -= danio;
    }

    public boolean estaVivo() {
        return vida > 0;
    }

    public void agregarEscudo(Escudo escudo) {
        this.escudo = escudo;
    }
}
